### PR TITLE
docs(lua): add docs for vim.lpeg, vim.re

### PR DESCRIPTION
**Pull request by EraldCaka**
addition of documentation for lpeg and re in lua.txt file for issue #26194

### CODE REVIEW

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -573,10 +573,180 @@ VIM.LPEG                                                            *lua-lpeg*
...
+    Returns a string with the running version of LPeg.
+
+vim.lpeg.setmaxstack(max)                                 *vim.lpeg.setmaxstack()*
+    Sets a limit for the size of the backtrack stack used by LPeg.
+
+    Parameters: ~
+      • {max} (integer) The maximum size of the stack.
+
+------------------------------------------------------------------------------
+VIM.RE                                                              *lua-lpeg*
```
Comment (clason) [inline runtime/doc/lua.txt:696]:
```suggestion
VIM.RE                                                              *lua-re*
```
Duplicate tag.

File: runtime/doc/lua.txt [hunk 2]
```diff
@@ -573,10 +573,180 @@ VIM.LPEG                                                            *lua-lpeg*
 
                                                              *vim.lpeg* *vim.re*
```
Comment (clason) [inline runtime/doc/lua.txt:574]:
```suggestion
                                                                     *vim.lpeg*
```

File: runtime/doc/lua.txt [hunk 3]
```diff
@@ -573,10 +573,180 @@ VIM.LPEG                                                            *lua-lpeg*
...
+    Returns a string with the running version of LPeg.
+
+vim.lpeg.setmaxstack(max)                                 *vim.lpeg.setmaxstack()*
+    Sets a limit for the size of the backtrack stack used by LPeg.
+
+    Parameters: ~
+      • {max} (integer) The maximum size of the stack.
+
+------------------------------------------------------------------------------
+VIM.RE                                                              *lua-re*
```
Comment (clason) [inline runtime/doc/lua.txt:696]:
```suggestion
VIM.RE                                                              *vim.re*
```
(and squash the commits; commit linter requires a space after the colon: `docs(lua): add docs for vim.lpeg, vim.re`)

Comment (EraldCaka) [reply runtime/doc/lua.txt:696]: okay let me fix that

**Comment by clason** [comment]
You do know how to squash commits, right?

**Comment by EraldCaka** [comment]
is it like gathering all your commits and pushing them all together?

**Comment by clason** [comment]
Yes. `git rebase -i HEAD~4`, `squash` the commits, then `git push -f`.

**Comment by EraldCaka** [comment]
image
okay did so

**Comment by clason** [comment]
What? No, you should replace all commits with a _single_ squashed commit.

**Comment by EraldCaka** [comment]
cleared it

**Comment by clason** [comment]
yes, but with the wrong commit message ;)

**Comment by EraldCaka** [comment]
let me change the commit message one second

**Comment by EraldCaka** [comment]
Done

**Review by justinmk — CHANGES_REQUESTED** [review]
Nope, not going to accept this unless it is generated. As suggested in https://github.com/neovim/neovim/issues/26194

We have already done lots of work to generate these Lua modules docs. This is now going in the reverse direction for a pretty weak reason (these modules are documented by their upstream).

**Pull request closed without merging**